src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,303 @@
+        End Function
+
+        Public Overrides Function InstrumentAddHandlerStatement(original As BoundAddHandlerStatement, rewritten As BoundStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.InstrumentAddHandlerStatement(original, rewritten))
+        End Function
+
+        Public Overrides Function InstrumentRemoveHandlerStatement(original As BoundRemoveHandlerStatement, rewritten As BoundStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.InstrumentRemoveHandlerStatement(original, rewritten))
+        End Function
+
+        Public Overrides Function InstrumentSyncLockObjectCapture(original As BoundSyncLockStatement, rewritten As BoundStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.InstrumentSyncLockObjectCapture(original, rewritten))
+        End Function
+
+        Public Overrides Function InstrumentWhileStatementConditionalGotoStart(original As BoundWhileStatement, ifConditionGotoStart As BoundStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.InstrumentWhileStatementConditionalGotoStart(original, ifConditionGotoStart))
+        End Function
+
+        Public Overrides Function InstrumentDoLoopStatementEntryOrConditionalGotoStart(original As BoundDoLoopStatement, ifConditionGotoStartOpt As BoundStatement) As BoundStatement
+            Dim previous As BoundStatement = MyBase.InstrumentDoLoopStatementEntryOrConditionalGotoStart(original, ifConditionGotoStartOpt)
+            If (original.ConditionOpt IsNot Nothing) Then
+                Return AddDynamicAnalysis(original, previous)
+            End If
+            Return previous
+        End Function
+
+        Public Overrides Function InstrumentIfStatementConditionalGoto(original As BoundIfStatement, condGoto As BoundStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.InstrumentIfStatementConditionalGoto(original, condGoto))
+        End Function
+
+        Public Overrides Function CreateSelectStatementPrologue(original As BoundSelectStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.CreateSelectStatementPrologue(original))
+        End Function
+
+        Public Overrides Function InstrumentFieldOrPropertyInitializer(original As BoundFieldOrPropertyInitializer, rewritten As BoundStatement, symbolIndex As Integer, createTemporary As Boolean) As BoundStatement
+            rewritten = MyBase.InstrumentFieldOrPropertyInitializer(original, rewritten, symbolIndex, createTemporary)
+            Dim syntax As VisualBasicSyntaxNode = original.Syntax
+
+            Select Case syntax.Parent.Parent.Kind()
+                Case SyntaxKind.VariableDeclarator, SyntaxKind.PropertyStatement
+                    Return AddDynamicAnalysis(original, rewritten)
+
+                Case Else
+                    Throw ExceptionUtilities.UnexpectedValue(syntax.Parent.Parent.Kind())
```

Comment:
If this case is unexpected, is the `Select Case` necessary or could we call `AddDynamicAnalysis` unconditionally?